src/helpers/slack.ts

```diff
@@ -10,29 +11,59 @@ import {
 const fs = require("fs");
 
 export function getEmojisToReactWith(text: string): Array<string> {
-    const emojisToReactWith: Array<string> = [];
+    const lowerCaseText = text.toLowerCase();
 
     // Read list of emojis from file
-    const rawdata = fs.readFileSync(
-        process.env.EMOJIS_FILE_PATH || "./helpers/files/emojis.json"
-    );
-    const emojis: Record<string, Array<string>> = JSON.parse(rawdata);
+    const emojis: Record<string, Array<string>> = readEmojisFromFile();
 
-    // flattens the array of keywords
     const keywords = Object.values(emojis).flat();
 
     // search for each keyword in the text
+    const emojisToReactWith = findMatchingEmojiKeywords(
+        keywords,
+        lowerCaseText,
+        emojis
+    );
+
+    return Array.from(new Set(emojisToReactWith));
+}
+
+function readEmojisFromFile(): Record<string, Array<string>> {
+    var emojisFilePath: string;
+    if (process.env.EMOJIS_FILE_PATH) {
+        // check if the file path is valid
+        if (!fs.existsSync(process.env.EMOJIS_FILE_PATH)) {
+            console.error(
+                `Emojis file path ${process.env.EMOJIS_FILE_PATH} is invalid.`
+            );
+            return {};
+        } else {
+            emojisFilePath = process.env.EMOJIS_FILE_PATH;
+        }
+    } else {
+        emojisFilePath = EMOJIS_FILE_PATH;
+    }
+    const rawdata = fs.readFileSync(emojisFilePath);
```

Comment:
[Non blocking comment]

May be wrap it around `try..catch` to ensure safer execution. Quick look at documentation shows that you can read data directly in `UTF-8` format instead of `bytes`. As current method assumes that data will be correctly encoded where as later would explicitly specify data should be read as raw strings.

Reply:
Ah nice, I have included the use of `try..catch` and reading data directly in UTF-8 format instead of bytes 👍